WooCommerce/Classes/ViewModels/ManualTrackingViewModel.swift

```diff
@@ -231,6 +238,12 @@ final class AddCustomTrackingViewModel: ManualTrackingViewModel {
     }
 
     let providerCellAccessoryType = UITableViewCell.AccessoryType.none
+    var isEmptyState: Bool {
+        (providerName.isNilOrEmpty && trackingNumber.isNilOrEmpty) ? true : false
+    }
+    var hasUnsavedChanges: Bool {
```

Comment:
If `providerName.isNilOrEmpty` is `true` and `trackingNumber.isNilOrEmpty` is also `true`, we would return `hasUnsavedChanges` to `true`, and I don't think that is correct. Am I right?

Reply:
That doesn't seem correct indeed, what do you think about something like the following?

```
    var hasUnsavedChanges: Bool {
        
        var returnValue = isEmptyState

        if returnValue {
            // Case isEmptyState = true -> hasUnsavedChanges = false
            returnValue = false
        } else if (providerName.isNilOrEmpty || trackingNumber.isNilOrEmpty) {
            // case providerName or trackingNumber are nilOrEmpty -> hasUnsavedChanges = true
            returnValue = true
        }
        
        return returnValue
    }
```